janitor/functions/select.py

```diff
@@ -339,23 +374,58 @@ def select(
 
     Args:
         df: A pandas DataFrame.
+        *args: Valid inputs include: an exact index name to look for,
+            a shell-style glob string (e.g. `*_thing_*`),
+            a regular expression,
+            a callable,
+            or variable arguments of all the aforementioned.
+            A sequence of booleans is also acceptable.
+            A dictionary can be used for selection
+            on a MultiIndex on different levels.
         rows: Valid inputs include: an exact label to look for,
             a shell-style glob string (e.g. `*_thing_*`),
             a regular expression,
             a callable,
             or variable arguments of all the aforementioned.
             A sequence of booleans is also acceptable.
-            A dictionary can be used for selection on a MultiIndex on different levels.
+            A dictionary can be used for selection
+            on a MultiIndex on different levels.
         columns: Valid inputs include: an exact label to look for,
             a shell-style glob string (e.g. `*_thing_*`),
             a regular expression,
             a callable,
             or variable arguments of all the aforementioned.
             A sequence of booleans is also acceptable.
-            A dictionary can be used for selection on a MultiIndex on different levels.
+            A dictionary can be used for selection
+            on a MultiIndex on different levels.
+        invert: Whether or not to invert the selection.
+            This will result in the selection
+            of the complement of the rows/columns provided.
+        axis: Should the selection be on the index('index'),
+            or columns('columns')?
+            Applicable only for the variable args parameter.
+
+    Raises:
+        ValueError: If args and rows and columns are provided.
 
     Returns:
         A pandas DataFrame with the specified rows and/or columns selected.
     """  # noqa: E501
 
-    return _select(df, args=None, rows=rows, columns=columns, axis="both")
+    if args:
+        check("invert", invert, [bool])
+        if (rows is not None) or (columns is not None):
+            raise ValueError(
+                "Either provide variable args with the axis parameter, "
+                "or provide arguments to the rows and/or columns parameters."
+            )
+        assert axis in {"index", "columns"}
```

Comment:
How about just putting this as an `else` condition below, rather than using an assert?

```python
if axis == "index": ...
elif axis == "columns": ...
else: raise ValueError()
```